src/rgw/rgw_data_sync.cc

```diff
@@ -5520,17 +5520,11 @@ int rgw_read_remote_bilog_info(const DoutPrefixProvider *dpp,
     { "info" , nullptr },
     { nullptr, nullptr }
   };
-  rgw_bucket_index_marker_info result;
-  int r = conn->get_json_resource(dpp, "/admin/log/", params, y, result);
+  int r = conn->get_json_resource(dpp, "/admin/log/", params, y, info);
   if (r < 0) {
     ldpp_dout(dpp, -1) << "failed to fetch remote log markers: " << cpp_strerror(r) << dendl;
     return r;
   }
-  r = markers.from_string(result.max_marker, -1);
```

Comment:
how is this change related to the generation fix?
you call "from_string" on the result of the function in 2 places. isn't it batter to keep it inside "rgw_read_remote_bilog_info()" ?

Reply:
the callers of `rgw_read_remote_bilog_info()` now need more than just `info.max_marker`, so i changed it to return `rgw_bucket_index_marker_info` directly instead of `BucketIndexShardsManager`. but i can change it to return both so the callers don't have to duplicate the marker parsing